go-app.js

```diff
@@ -69,18 +68,104 @@ go.app = function() {
                 characters_per_page: 160,
                 options_per_page : 3,
                 next: function(choice) {
-                    return 'states:end';
+                    return { 
+                        name: 'states:retrieve-location',
+                        creator_opts : {
+                            full_geoid : choice.value,
+                            full_name : choice.label
+                        }
+                    };
                 }   
             });
         });
-      
+
+        self.states.add('states:retrieve-location', function(name, opts) {
+            return self
+                .http.get('http://wazimap.co.za/profiles/' + opts.full_geoid + '.json')
+                .then(function(response) {
+                    opts.data = response.data;
+                    return  self.states.create(
+                        'states:select-section', opts);
+                });
+        });
+
+        self.states.add('states:select-section', function(name, opts) {
+            return new ChoiceState(name, {
+                question: 'I would like to query:',
+                choices: [
+                    new Choice('elections', 'Elections'),
+                    new Choice ('demographics', 'Demographics'),
+                    new Choice('households', 'Households'),
+                    new Choice('service_delivery', 'Service Delivery'),
+                    new Choice('economics', 'Economics'),
+                    new Choice('education', 'Education'),
+                    new Choice('children', 'Children'),
+                    new Choice('child_households', 'Child-headed Households')
+                        ],
+                    next: function(choice) {
```

Comment:
I think `next:` is indented too far (should start at the same level as `choices:` and the `],` should also start at the same level (more nitpicking).